src/index.js

```diff
@@ -68,66 +66,70 @@ const mapToPropertyDescriptor = (
           return null;
         };
       },
-      set: function(value) {
-        eventHandler = (typeof value === 'function') ? value : null;
-        onAfterSet(this);
+      set(value) {
+        this[name] = (typeof value === 'function') ? value : null;
       }
     };
-  } else if (typeOrSerDes === Types.bool) {
+  }
+
+  // booleans
+  if (typeOrSerDes === Types.bool) {
     return {
-      get: function() {
+      get() {
         return this.hasAttribute(name);
       },
-      set: function(value) {
+      set(value) {
         if (value) {
           this.setAttribute(name, '');
-        } else {
-          this.removeAttribute(name);
+          return;
         }
-        onAfterSet(this);
+
+        this.removeAttribute(name);
       }
     };
-  } else {
-    return {
-      get: function() {
-        const value = this.getAttribute(name);
+  }
 
-        if (typeOrSerDes === Types.number) {
-          return Number(value);
-        } else if (typeOrSerDes === Types.json) {
-          return JSON.parse(value);
-        }
+  // string, numbers, json
+  return {
+    get() {
+      const value = this.getAttribute(name);
 
-        return (typeof typeOrSerDes.deserialize === 'function')
-          ? typeOrSerDes.deserialize(value)
-          : value;
-      },
-      set: function(value) {
-        const attributeValue = (() => {
-          if (typeOrSerDes === Types.json) {
-            return JSON.stringify(value);
-          }
-
-          return (typeof typeOrSerDes.serialize === 'function')
-            ? typeOrSerDes.serialize(value)
-            : value.toString();
-        })();
-
-        this.setAttribute(name, attributeValue);
-        onAfterSet(this);
+      if (typeOrSerDes === Types.number) {
+        return Number(value);
       }
-    };
-  }
+
+      if (typeOrSerDes === Types.json) {
+        return JSON.parse(value);
```

Comment:
Yeah I was thinking the same thing. it would be pretty bad to get json parse errors from your web component